### PR TITLE
XNNPACK backend supports subgraph API

### PR DESCRIPTION
**Pull request by huningxin**
XNNPACK subgraph API is used by [TFLite XNNPACK delegate](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/delegates/xnnpack) for graph level optimization and execution with good performance.

This PR updates WebNN-native's XNNPACK backend to support subgraph API with 27 ops support
* binary including add, sub, mul, div, max, min
* clamp
* concat (only support concatenation of up to 4 inputs)
* conv2d (only support nhwc layout)
* gemm (only support constant weights)
* matmul (only support 2d operand b)
* pad
* pool2d: average and max (only support nhwc layout)
* reshape
* split (only support even split up to 4 outputs)
* squeeze
* unary including abs, ceil, floor, neg
* activations: relu, leakyRelu, hardswish, sigmoid
* softmax

webnn_end2end_tests results:
[  PASSED  ] 103 tests including MobileNetV2Nhwc and SqueezeNetNhwc
267 FAILED TESTS

@fujunwei , PTAL.

This PR also adds constant weights cases in gemm tests, @BruceDai , please also take a look.

### CODE REVIEW

File: src/webnn_native/WebnnNative.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ namespace webnn_native {
     namespace {
 
         void DumpMemoryLeaks() {
-#if defined(_WIN32) && defined(_DEBUG)
+#if 0
```
Comment (fujunwei) [inline src/webnn_native/WebnnNative.cpp:34]: Do we disable the memory leak check?

File: src/webnn_native/xnnpack/GraphXNN.h [hunk 2]
```diff
@@ -24,17 +24,22 @@
...
 #include "webnn_native/ops/Pool2d.h"
 #include "webnn_native/ops/Reshape.h"
+#include "webnn_native/ops/Split.h"
+#include "webnn_native/ops/Squeeze.h"
 #include "webnn_native/ops/Transpose.h"
 #include "webnn_native/ops/Unary.h"
 #include "webnn_native/xnnpack/ContextXNN.h"
 
-namespace webnn_native::xnnpack {
+namespace webnn_native { namespace xnnpack {
```
Comment (fujunwei) [inline src/webnn_native/xnnpack/GraphXNN.h:42]: should we use nested namespace?

**Review by fujunwei — APPROVED** [review]
LGTM with some comments.

File: src/webnn_native/WebnnNative.cpp (hunk 1 above)
Comment (huningxin) [reply src/webnn_native/WebnnNative.cpp:34]: My bad. I'll revert it.

File: src/webnn_native/xnnpack/GraphXNN.h (hunk 2 above)
Comment (huningxin) [reply src/webnn_native/xnnpack/GraphXNN.h:42]: we should, i'll fix that.

File: src/webnn_native/WebnnNative.cpp (hunk 1 above)
Comment (huningxin) [reply src/webnn_native/WebnnNative.cpp]: fixed.

File: src/webnn_native/xnnpack/GraphXNN.h (hunk 2 above)
Comment (huningxin) [reply src/webnn_native/xnnpack/GraphXNN.h]: fixed, please take another look.

**Comment by fujunwei** [comment]
LGTM, we can merge it when the CI is green.

**Comment by huningxin** [comment]
CI passed. I am going to merge it. Thanks for the review!
